test/litmus_test.exs

```diff
@@ -38,7 +38,10 @@ defmodule LitmusTest do
           min_length: 2,
           max_length: 5
         },
-        "start_date" => %Litmus.Type.DateTime{}
+        "start_date" => %Litmus.Type.DateTime{},
+        "email" => %Litmus.Type.String{
+          default: %Litmus.Type.Any.Default{value: ""}
```

Comment:
I'm wondering if it's simpler to just default without a specific `Litmus.Type.Any.Default` type.

For example:

```elixir
"email" => %Litmus.Type.String{
  default: ""
}
```

That way you can get rid of `lib/litmus/type/any/default.ex` entirely too.

It doesn't seem like the full type is needed because it only has a single setting, `value`, and you don't match on it here: https://github.com/lob/litmus/pull/22/files#diff-f3f0646ef19d1731da8324f6e750008dR5

Reply:
I agree that it would be better to just pass a bare value, however I tried that originally, and there is a problem handling the case when we want `nil` default values.

Suppose we did use a raw value for the `default` key. In the struct definition, we'd have something like the following:

```elixir
defmodule Litmus.Type.String do
  ...
  defstruct [
    :default,
    :min_length,
    :max_length,
    :length,
    regex: %Type.String.Regex{},
    trim: false,
    required: false
  ]
  ...
end
```

This means that the default value for all strings is set to `nil`. Then if we had a situation like the following:

```elixir
schema = %{
  "id" => %Litmus.Type.String{},
  "email" => %Litmus.Type.String{}
}

params = %{}

Litmus.validate(params, schema) # %{id: nil, email: nil}
```

Where each of the values would be populated with `nil`. I wasn't sure if this was desirable behavior, as we are adding a default without the user explicitly setting one. 

I thought about just adding a check to `Litmus.Default.validate_field/3` to set defaults when the `default` key isn't `nil`:

```elixir
defmodule Litmus.Default do
  @moduledoc false

  @spec validate(map, String.t(), map) :: {:ok, map}
  def validate(%{default: default_value}, field, params) when default_value != nil do
    {:ok, Map.put_new(params, field, default_value)}
  end

  def validate(_, _, params), do: {:ok, params}
end
```

but this excludes the case where we want the default value to actually be nil.

Are we alright just having schema fields default to `nil`? Or is there another way we could handle bare default values?

Reply:
@mgartner just wanted to bump this ☝️

Reply:
The only alternative I can think of is to set the value of `default` to something that will be unlikely to ever be used, like `Litmus.Type.Any.NoDefault` (as-in the atom), in each type struct, like:

```elixir
defmodule Litmus.Type.String do
  ...
  defstruct [
    default: Litmus.Type.Any.NoDefault,
    :min_length,
    :max_length,
    :length,
    regex: %Type.String.Regex{},
    trim: false,
    required: false
  ]
  ...
end
```

But I'm not sure if that's better or worse. Thoughts?

@shraddha1704 @kylekwong what do you think?

Reply:
I like that idea better 👍 There really shouldn't be any reason that the `Litmus.Type.Any.NoDefault` atom would be used, so as long as the values for `default` are anything but that, we could handle it easily in `Litmus.Default.validate/3`. If you're all alright with that too I can make those changes.

Reply:
I like the idea too, but isn't the term `NoDefault` confusing here?

Reply:
@shraddha1704 users shouldn't be setting that value. It should only be in the Litmus library. If they want to set a default then the specify it, but the value of the `default` field is `Litmus.Type.Any.NoDefault` unless the user specifies differently.

Reply:
@mgartner Thanks. I guess NoDefault is basically NoUserDefinedDefaults?

Reply:
yes. i'm open to other names too, but that's the idea

Reply:
I'll push up the changes 👍